src/commands/vote-mockup.ts

```diff
@@ -44,30 +60,32 @@ export default class VoteMockupCommand extends SlashCommand<Client> {
 
     await ctx.fetch();
 
-    const ballot = new Map<string, string>();
-
     ctx.registerComponent('vote', async (selectCtx) => {
-      if (game.players.findIndex((p) => p.id === selectCtx.user.id) === -1) {
-        selectCtx.send('You are not in this game.', { ephemeral: true });
+      console.log(`Received vote with ${start + duration - Date.now()}ms remaining`);
+
+      if (game.players.findIndex((p) => p.id === selectCtx.user.id) === -1)
+        return selectCtx.send('You are not in this game.', { ephemeral: true });
+
+      if (start + duration < Date.now()) {
+        return selectCtx.send('The vote has ended.', { ephemeral: true });
       }
 
       const currentVote = ballot.get(selectCtx.user.id);
       const newVote = selectCtx.values[0];
 
       if (currentVote === newVote) {
-        await selectCtx.send('Unchanged.', { ephemeral: true });
-        return;
+        return selectCtx.send('Unchanged.', { ephemeral: true });
       }
 
       ballot.set(selectCtx.user.id, newVote);
       await selectCtx.send(`You have voted for <@${newVote}>.`, { ephemeral: true });
 
       await selectCtx.editParent({
-        content: `Vote mockup (${ballot.size} / ${game.players.length})`
+        embeds: [getBallotEmbed()]
       });
     });
 
-    await wait(1000 * 20); // 20 seconds
+    await wait(duration - 1000); // 20 seconds minus 1 second for the wait - idk why
```

Comment:
As stated before, I don't know why it has significant delay - except the possibility of network problems during testing.

While it may be a stretch to suggest this with the physical complications of Discord's application structure (as well as the legal concerns), it may be possible to build a mock interface for bots to communicate with as a development environment (everything from requesting a user account to creating a message). _It is understandably quite a mammoth of a task to grasp, but would be a valued asset to general development if something remotely like it can be achieved for interactions as a whole._